x-pack/plugins/stack_connectors/server/connector_types/lib/convert_timestamp.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export function convertTimestamp(timestamp?: string | number | null): string | number | null {
+  if (timestamp) {
+    if (typeof timestamp === 'string') {
+      const trimmedTimestamp = timestamp.trim();
+      if (trimmedTimestamp.length > 0) {
+        const parsedTimestamp = parseInt(trimmedTimestamp, 10);
+
+        if (!isNaN(parsedTimestamp) && JSON.stringify(parsedTimestamp) === trimmedTimestamp) {
```

Comment:
I guess this is for the case of `123_but_not_a_date` not returning 123?  Good catch!

Note however:

    parseInt('123.0') === 123

So that would fail.  I think I usually just deal with the prefix, don't do that sort of check.

Poked around the Kibana repo looking at other `parseInt()` calls - trying to find some parsing "human" data - and I'm not seeing anyone else making other checks.  Of course, folks may not generally know that `parseInt()` will stop once it hits a non-numeric character and return what it read up to that.

Do you think there's a need for us to do extra checking here?

Reply:
Yes you are right it is to avoid converting the strings starting with numbers.

For date inputs we should only convert the epoch number passed as string, all the rest should remain as string.
Epoch numbers are always integer so `JSON.stringify(parsedTimestamp) === trimmedTimestamp` check guarantees that.

`'123.0'` would remain as `string '123.0'`  and moment validation would return false for it, as expected.

And if we don't do that check `2024-10-10` (valid date string) would be converted to `2024` and lead us to a bug :)